internal/controller/sink_test.go

```diff
@@ -118,6 +118,12 @@ func TestConfigSink_syncAIGatewayRoute(t *testing.T) {
 		// Defaulting to the first backend.
 		require.Equal(t, "some-backend1", string(updatedHTTPRoute.Spec.Rules[2].BackendRefs[0].BackendRef.Name))
 		require.Equal(t, "/", *updatedHTTPRoute.Spec.Rules[2].Matches[0].Path.Value)
+		// Status should be one and is about reconciling
+		require.Len(t, route.Status.Conditions, 1)
+		require.Equal(t, ConditionReconciled, route.Status.Conditions[0].Type)
```

Comment:
It might be a personal preference but unit tests should verify expected behavior rather than mirror the implementation. Using constants from the main code in assertions can lead to false positives and reduce the test’s effectiveness. Instead, define expected values explicitly in the test to ensure failures reflect real issues.

Reply:
Sure, just ensure the Status is True.